### PR TITLE
[FLINK-29907][Connectors/AWS] Externalize KDS/KDF and AWS base from Flink repo

### PR DESCRIPTION
**Pull request by dannycranmer**
This change externalizes the AWS base, Firehose and Kinesis connector modules from the Flink repository from [8a47420adfe9af7f1c303874c7a32dab3229ea66](https://github.com/apache/flink/commit/8a47420adfe9af7f1c303874c7a32dab3229ea66).

During this migration the following dependency changes have been made to KDS/KDF:
- AWS SDK Netty dependencies upgraded from `4.1.70.Final` to `4.1.77.Final`
  - io.netty:netty-buffer
  - io.netty:netty-codec-http2
  - io.netty:netty-codec-http
  - io.netty:netty-codec
  - io.netty:netty-common
  - io.netty:netty-handler
  - io.netty:netty-resolver
  - io.netty:netty-transport-classes-epoll
  - io.netty:netty-transport-native-unix-common
  - io.netty:netty-transport 
- `joda-time` upgraded from `2.5.0` > `2.8.1`

The DynamoDB connector has taken the following changes to align with the other/Flink:
- `commons-logging` downgraded from `1.2` to `1.1.3`
- `httpcomponents:httpcore` upgraded from `4.4.13` to `4.4.14`
- `commons-codec:commons-codec` upgraded from `1.11` to `1.15`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -102,6 +109,50 @@ under the License.
...
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>${mockito.version}</version>
+            <type>jar</type>
+            <scope>test</scope>
+        </dependency>
```
Comment (hlteoh37) [inline pom.xml:132]: 😭
Shall we create a followup JIRA to remove this? Seems like used in quite a few places though...

File: flink-sql-connector-kinesis/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- commons-lang:commons-lang:2.6
+- commons-logging:commons-logging:1.1.3
+- commons-codec:commons-codec:1.15
+- org.apache.commons:commons-lang3:3.3.2
+- com.google.guava:guava:29.0-jre
+- com.google.guava:failureaccess:1.0.1
+- com.fasterxml.jackson.core:jackson-annotations:2.13.4
+- com.fasterxml.jackson.core:jackson-databind:2.13.4.2
+- com.fasterxml.jackson.core:jackson-core:2.13.4
+- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.13.4
```
Comment (hlteoh37) [inline flink-sql-connector-kinesis/src/main/resources/META-INF/NOTICE:20]: Verified with CI build
```

[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-sql-connector-kinesis ---
[INFO] Including org.apache.flink:flink-connector-kinesis:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including joda-time:joda-time:jar:2.8.1 in the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.36 from the shaded jar.
[INFO] Including commons-io:commons-io:jar:2.11.0 in the shaded jar.
[INFO] Including commons-lang:commons-lang:jar:2.6 in the shaded jar.
[INFO] Excluding jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2 from the shaded jar.
[INFO] Excluding jakarta.activation:jakarta.activation-api:jar:1.2.1 from the shaded jar.
[INFO] Including org.apache.commons:commons-lang3:jar:3.3.2 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.1.3 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-core:jar:2.13.4 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-databind:jar:2.13.4.2 in the shaded jar.
[INFO] Including com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4 in the shaded jar.
[INFO] Including com.google.guava:guava:jar:29.0-jre in the shaded jar.
[INFO] Including com.google.guava:failureaccess:jar:1.0.1 in the shaded jar.
[INFO] Excluding com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava from the shaded jar.
[INFO] Excluding com.google.code.findbugs:jsr305:jar:1.3.9 from the shaded jar.
[INFO] Excluding org.checkerframework:checker-qual:jar:2.11.1 from the shaded jar.
[INFO] Excluding com.google.errorprone:error_prone_annotations:jar:2.3.4 from the shaded jar.
[INFO] Excluding com.google.j2objc:j2objc-annotations:jar:1.3 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-guava:jar:30.1.1-jre-15.0 from the shaded jar.
[INFO] Including org.apache.flink:flink-connector-base:jar:1.16.0 in the shaded jar.
[INFO] Excluding org.apache.flink:flink-table-common:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-annotations:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-metrics-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-asm-9:jar:9.2-15.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-jackson:jar:2.12.4-15.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.kryo:kryo:jar:2.24.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.minlog:minlog:jar:1.2 from the shaded jar.
[INFO] Excluding commons-collections:commons-collections:jar:3.2.2 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-compress:jar:1.21 from the shaded jar.
[INFO] Excluding com.ibm.icu:icu4j:jar:67.1 from the shaded jar.
[INFO] Including commons-codec:commons-codec:jar:1.15 in the shaded jar.
[INFO] Including com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.13.4 in the shaded jar.
[INFO] Excluding org.objenesis:objenesis:jar:2.1 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-force-shading:jar:16.0 from the shaded jar.
[INFO] No artifact matching filter io.netty:netty
```

File: flink-sql-connector-aws-kinesis-streams/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -0,0 +1,49 @@
+flink-sql-connector-aws-kinesis-streams
+
+Copyright 2014-2022 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+- software.amazon.awssdk:kinesis:2.17.247
```
Comment (hlteoh37) [inline flink-sql-connector-aws-kinesis-streams/src/main/resources/META-INF/NOTICE:10]: LGTM. Verified from CI
```
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ flink-sql-connector-aws-kinesis-streams ---
[INFO] Building jar: /home/runner/work/flink-connector-aws/flink-connector-aws/flink-sql-connector-aws-kinesis-streams/target/flink-sql-connector-aws-kinesis-streams-4.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-sql-connector-aws-kinesis-streams ---
[INFO] Including org.apache.flink:flink-connector-aws-kinesis-streams:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-base:jar:1.16.0 in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-aws-base:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including software.amazon.awssdk:netty-nio-client:jar:2.17.247 in the shaded jar.
[INFO] Including io.netty:netty-codec-http:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec-http2:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-resolver:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-buffer:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-handler:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-classes-epoll:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-native-unix-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including org.reactivestreams:reactive-streams:jar:1.0.3 in the shaded jar.
[INFO] Including software.amazon.awssdk:sts:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-query-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:kinesis:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-cbor-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-dataformat-cbor:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-json-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:json-utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:protocol-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:profiles:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:sdk-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:auth:jar:2.17.247 in the shaded jar.
[INFO] Excluding software.amazon.eventstream:eventstream:jar:1.0.1 from the shaded jar.
[INFO] Including software.amazon.awssdk:http-client-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:regions:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:annotations:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:metrics-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:apache-client:jar:2.17.247 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpclient:jar:4.5.13 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.1.3 in the shaded jar.
[INFO] Excluding commons-codec:commons-codec:jar:1.15 from the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore:jar:4.4.14 in the shaded jar.
[INFO] Excluding org.apache.flink:flink-table-common:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-asm-9:jar:9.2-15.0 from the shaded jar.
[INFO] Excluding com.ibm.icu:icu4j:jar:67.1 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-force-shading:jar:16.0 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-compress:jar:1.21 from the shaded jar.
[INFO] Excluding org.objenesis:objenesis:jar:2.1 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-guava:jar:30.1.1-jre-15.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-jackson:jar:2.12.4-15.0 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-lang3:jar:3.3.2 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-annotations:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-metrics-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.kryo:kryo:jar:2.24.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.minlog:minlog:jar:1.2 from the shaded jar.
[INFO] Excluding commons-collections:commons-collections:jar:3.2.2 from the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.36 from the shaded jar.
[INFO] Excluding com.google.code.findbugs:jsr305:jar:1.3.9 from the shaded jar.
[INFO] No artifact matching filter io.netty:netty

```

File: flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE [hunk 4]
```diff
@@ -0,0 +1,47 @@
+flink-sql-connector-aws-kinesis-firehose
+
+Copyright 2014-2022 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+This project bundles the following dependencies under the Apache Software License 2.0. (http://www.apache.org/licenses/LICENSE-2.0.txt)
+
+- software.amazon.awssdk:firehose:2.17.247
```
Comment (hlteoh37) [inline flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE:10]: LGTM. Verified
```

[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-sql-connector-aws-kinesis-firehose ---
[INFO] Including org.apache.flink:flink-connector-aws-kinesis-firehose:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-base:jar:1.16.0 in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-aws-base:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including software.amazon.awssdk:sts:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-query-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:profiles:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:firehose:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-json-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:json-utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:protocol-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:sdk-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:auth:jar:2.17.247 in the shaded jar.
[INFO] Excluding software.amazon.eventstream:eventstream:jar:1.0.1 from the shaded jar.
[INFO] Including software.amazon.awssdk:http-client-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:regions:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:annotations:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:metrics-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:apache-client:jar:2.17.247 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpclient:jar:4.5.13 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.1.3 in the shaded jar.
[INFO] Excluding commons-codec:commons-codec:jar:1.15 from the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore:jar:4.4.14 in the shaded jar.
[INFO] Including software.amazon.awssdk:netty-nio-client:jar:2.17.247 in the shaded jar.
[INFO] Including io.netty:netty-codec-http:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec-http2:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-resolver:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-buffer:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-handler:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-classes-epoll:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-native-unix-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including org.reactivestreams:reactive-streams:jar:1.0.3 in the shaded jar.
[INFO] Excluding org.apache.flink:flink-table-common:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-asm-9:jar:9.2-15.0 from the shaded jar.
[INFO] Excluding com.ibm.icu:icu4j:jar:67.1 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-force-shading:jar:16.0 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-compress:jar:1.21 from the shaded jar.
[INFO] Excluding org.objenesis:objenesis:jar:2.1 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-guava:jar:30.1.1-jre-15.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-jackson:jar:2.12.4-15.0 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-lang3:jar:3.3.2 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-annotations:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-metrics-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.kryo:kryo:jar:2.24.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.minlog:minlog:jar:1.2 from the shaded jar.
[INFO] Excluding commons-collections:commons-collections:jar:3.2.2 from the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.36 from the shaded jar.
[INFO] Excluding com.google.code.findbugs:jsr305:jar:1.3.9 from the shaded jar.
```

File: flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- io.netty:netty-codec:4.1.77.Final
+- io.netty:netty-transport:4.1.77.Final
+- io.netty:netty-resolver:4.1.77.Final
+- io.netty:netty-common:4.1.77.Final
+- io.netty:netty-buffer:4.1.77.Final
+- io.netty:netty-handler:4.1.77.Final
+- io.netty:netty-transport-native-unix-common:4.1.77.Final
+- io.netty:netty-transport-classes-epoll:4.1.77.Final
+- com.typesafe.netty:netty-reactive-streams-http:2.0.5
+- com.typesafe.netty:netty-reactive-streams:2.0.5
```
Comment (hlteoh37) [inline flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE:39]: These don't seem to be in the shaded output

File: flink-sql-connector-dynamodb/src/main/resources/META-INF/NOTICE [hunk 6]
```diff
@@ -37,9 +37,9 @@ This project bundles the following dependencies under the Apache Software Licens
 - io.netty:netty-transport-native-unix-common:4.1.77.Final
 - io.netty:netty-transport-classes-epoll:4.1.77.Final
 - org.apache.httpcomponents:httpclient:4.5.13
-- org.apache.httpcomponents:httpcore:4.4.13
-- commons-logging:commons-logging:1.2
-- commons-codec:commons-codec:1.11
+- org.apache.httpcomponents:httpcore:4.4.14
+- commons-logging:commons-logging:1.1.3
+- commons-codec:commons-codec:1.15
```
Comment (hlteoh37) [inline flink-sql-connector-dynamodb/src/main/resources/META-INF/NOTICE:42]: LGTM. Verified from CI build
```
[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-sql-connector-dynamodb ---
[INFO] Including org.apache.flink:flink-connector-dynamodb:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-base:jar:1.16.0 in the shaded jar.
[INFO] Including org.apache.flink:flink-connector-aws-base:jar:1.16.0 in the shaded jar.
[INFO] Including software.amazon.awssdk:netty-nio-client:jar:2.17.247 in the shaded jar.
[INFO] Including io.netty:netty-codec-http:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec-http2:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-resolver:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-buffer:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-handler:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-classes-epoll:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-native-unix-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including org.reactivestreams:reactive-streams:jar:1.0.3 in the shaded jar.
[INFO] Including software.amazon.awssdk:sts:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-query-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:dynamodb:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-json-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:json-utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:protocol-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:profiles:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:sdk-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:auth:jar:2.17.247 in the shaded jar.
[INFO] Excluding software.amazon.eventstream:eventstream:jar:1.0.1 from the shaded jar.
[INFO] Including software.amazon.awssdk:http-client-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:regions:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:annotations:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:metrics-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:apache-client:jar:2.17.247 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpclient:jar:4.5.13 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.1.3 in the shaded jar.
[INFO] Including commons-codec:commons-codec:jar:1.15 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore:jar:4.4.14 in the shaded jar.
[INFO] Including software.amazon.awssdk:dynamodb-enhanced:jar:2.17.247 in the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-force-shading:jar:16.0 from the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.36 from the shaded jar.
[INFO] No artifact matching filter io.netty:netty
[INFO] Replacing original artifact with shaded artifact.
```

File: flink-connector-kinesis/src/main/resources/META-INF/NOTICE [hunk 7]
```diff
@@ -0,0 +1,293 @@
+flink-connector-kinesis
+Copyright 2014-2022 The Apache Software Foundation
+
+This product includes software developed at
```
Comment (hlteoh37) [inline flink-connector-kinesis/src/main/resources/META-INF/NOTICE:4]: LGTM

```
[INFO] --- maven-shade-plugin:3.1.1:shade (shade-flink) @ flink-connector-kinesis ---
[INFO] Including com.amazonaws:aws-java-sdk-kinesis:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-core:jar:1.12.276 in the shaded jar.
[INFO] Including software.amazon.ion:ion-java:jar:1.0.2 in the shaded jar.
[INFO] Excluding joda-time:joda-time:jar:2.8.1 from the shaded jar.
[INFO] Including com.amazonaws:jmespath-java:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-sts:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-kms:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-s3:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-dynamodb:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:aws-java-sdk-cloudwatch:jar:1.12.276 in the shaded jar.
[INFO] Including com.amazonaws:amazon-kinesis-producer:jar:0.14.1 in the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.36 from the shaded jar.
[INFO] Including com.google.protobuf:protobuf-java:jar:3.11.4 in the shaded jar.
[INFO] Excluding commons-io:commons-io:jar:2.11.0 from the shaded jar.
[INFO] Excluding commons-lang:commons-lang:jar:2.6 from the shaded jar.
[INFO] Excluding jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.2 from the shaded jar.
[INFO] Excluding jakarta.activation:jakarta.activation-api:jar:1.2.1 from the shaded jar.
[INFO] Including com.amazonaws:amazon-kinesis-client:jar:1.14.8 in the shaded jar.
[INFO] Excluding org.apache.commons:commons-lang3:jar:3.3.2 from the shaded jar.
[INFO] Excluding commons-logging:commons-logging:jar:1.1.3 from the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-core:jar:2.13.4 from the shaded jar.
[INFO] Including com.amazonaws:dynamodb-streams-kinesis-adapter:jar:1.5.3 in the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-databind:jar:2.13.4.2 from the shaded jar.
[INFO] Excluding com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4 from the shaded jar.
[INFO] Excluding com.google.guava:guava:jar:29.0-jre from the shaded jar.
[INFO] Excluding com.google.guava:failureaccess:jar:1.0.1 from the shaded jar.
[INFO] Excluding com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava from the shaded jar.
[INFO] Excluding com.google.code.findbugs:jsr305:jar:1.3.9 from the shaded jar.
[INFO] Excluding org.checkerframework:checker-qual:jar:2.11.1 from the shaded jar.
[INFO] Excluding com.google.errorprone:error_prone_annotations:jar:2.3.4 from the shaded jar.
[INFO] Excluding com.google.j2objc:j2objc-annotations:jar:1.3 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-guava:jar:30.1.1-jre-15.0 from the shaded jar.
[INFO] Including org.apache.flink:flink-connector-aws-kinesis-streams:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Excluding org.apache.flink:flink-connector-base:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-table-common:jar:1.16.0 from the shaded jar.
[INFO] Including org.apache.flink:flink-connector-aws-base:jar:4.0-SNAPSHOT in the shaded jar.
[INFO] Excluding org.apache.flink:flink-annotations:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-metrics-core:jar:1.16.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-asm-9:jar:9.2-15.0 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-jackson:jar:2.12.4-15.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.kryo:kryo:jar:2.24.0 from the shaded jar.
[INFO] Excluding com.esotericsoftware.minlog:minlog:jar:1.2 from the shaded jar.
[INFO] Excluding commons-collections:commons-collections:jar:3.2.2 from the shaded jar.
[INFO] Excluding org.apache.commons:commons-compress:jar:1.21 from the shaded jar.
[INFO] Excluding com.ibm.icu:icu4j:jar:67.1 from the shaded jar.
[INFO] Including software.amazon.awssdk:kinesis:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-cbor-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-dataformat-cbor:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:third-party-jackson-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-json-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:json-utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:protocol-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:profiles:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:sdk-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:auth:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.eventstream:eventstream:jar:1.0.1 in the shaded jar.
[INFO] Including software.amazon.awssdk:http-client-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:regions:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:annotations:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:utils:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-core:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:metrics-spi:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:apache-client:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:netty-nio-client:jar:2.17.247 in the shaded jar.
[INFO] Including io.netty:netty-codec-http:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec-http2:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-codec:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-resolver:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-buffer:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-handler:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-classes-epoll:jar:4.1.77.Final in the shaded jar.
[INFO] Including io.netty:netty-transport-native-unix-common:jar:4.1.77.Final in the shaded jar.
[INFO] Including org.reactivestreams:reactive-streams:jar:1.0.3 in the shaded jar.
[INFO] Including software.amazon.awssdk:sts:jar:2.17.247 in the shaded jar.
[INFO] Including software.amazon.awssdk:aws-query-protocol:jar:2.17.247 in the shaded jar.
[INFO] Including org.apache.httpcomponents:httpclient:jar:4.5.13 in the shaded jar.
[INFO] Excluding commons-codec:commons-codec:jar:1.15 from the shaded jar.
[INFO] Including org.apache.httpcomponents:httpcore:jar:4.4.14 in the shaded jar.
[INFO] Excluding com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.13.4 from the shaded jar.
[INFO] Excluding org.apache.flink:flink-shaded-force-shading:jar:16.0 from the shaded jar.
[INFO] Excluding org.objenesis:objenesis:jar:2.1 from the shaded jar
[INFO] No artifact matching filter io.netty:netty
```

**Review by hlteoh37 — APPROVED** [review]
LGTM, have a nit on 1 notice file

File: flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE (hunk 5 above)
Comment (dannycranmer) [reply flink-sql-connector-aws-kinesis-firehose/src/main/resources/META-INF/NOTICE:39]: Good spot. I will remove them Thanks

**Pull request merged**